dotCMS/src/main/java/com/dotcms/rest/api/v1/storage/ContentStorageResource.java

```diff
@@ -0,0 +1,218 @@
+import java.util.List;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+/**
+ * Resource to expose the storage and metadata
+ * @author jsanca
+ */
+@Path("/v1/storage")
+public class ContentStorageResource {
+
+    private final WebResource webResource;
+    private final FileStorageAPI fileStorageAPI;
+    private final MultiPartUtils multiPartUtils;
+    private final ContentStorageHelper helper;
+    private final ContentletMetadataAPI contentletMetadataAPI;
+    private final LanguageWebAPI languageWebAPI;
+
+    @VisibleForTesting
+    ContentStorageResource(final FileStorageAPI fileStorageAPI,
+            final WebResource webResource,
+            final MultiPartUtils multiPartUtils,
+            final ContentStorageHelper helper,
+            final ContentletMetadataAPI contentletMetadataAPI, final LanguageWebAPI languageWebAPI) {
+
+        this.fileStorageAPI = fileStorageAPI;
+        this.webResource    = webResource;
+        this.multiPartUtils = multiPartUtils;
+        this.helper = helper;
+        this.contentletMetadataAPI = contentletMetadataAPI;
+        this.languageWebAPI = languageWebAPI;
+    }
+
+    public ContentStorageResource() {
+        this(APILocator.getFileStorageAPI(), new WebResource(),
+                new MultiPartUtils(), new ContentStorageHelper(),
+                APILocator.getContentletMetadataAPI(), WebAPILocator.getLanguageWebAPI());
+    }
+
+    @GET
+    @JSONP
+    @NoCache
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public final Response getFileFromStorage(@Context final HttpServletRequest request,
+                                           @Context final HttpServletResponse response,
+                                           @QueryParam("path") final String path) {
+
+        final File   file        = StoragePersistenceProvider.INSTANCE.get().getStorage().pullFile("files", path);
+        final String fileName    = FilenameUtils.getName(path);
+
+        return Response.ok(
+                new StreamingOutput() {
+                    @Override
+                    public void write(OutputStream output)  {
+                        try (ZipOutputStream zoutput = new ZipOutputStream(output);
+                                InputStream fileInput = Files.newInputStream(file.toPath())) {
+
+                            int   bytesRead      = 0;
+                            final byte[] buffer  = new byte[512];
+                            zoutput.putNextEntry(new ZipEntry(path));
+
+                            while ((bytesRead = fileInput.read(buffer)) > 0) {
+
+                                zoutput.write(buffer, 0, bytesRead);
+                            }
+
+                            zoutput.closeEntry();
+                        } catch (Exception e) {
+
+                            Logger.warn(this.getClass(), e.getMessage(), e);
+                            throw new DotRuntimeException(e.getMessage());
```

Comment:
Codacy found an issue: [New exception is thrown in catch block, original stack trace may be lost](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)